core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackStopRestrictionService.java

```diff
@@ -58,19 +95,74 @@ public StopRestrictionReason isInfrastructureStoppable(Stack stack) {
         return reason;
     }
 
+    @VisibleForTesting
+    boolean isEphemeralInstanceGroupStoppable(Set<ServiceComponent> serviceComponents) {
+        for (ServiceRoleGroup serviceRoleGroup: config.getPermittedServiceRoleGroups()) {
+            if (areServiceComponentsPermitted(serviceRoleGroup, serviceComponents) && areRequiredServiceRolesPresent(serviceRoleGroup, serviceComponents)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private boolean areServiceComponentsPermitted(ServiceRoleGroup serviceRoleGroup, Set<ServiceComponent> serviceComponents) {
+        Set<ServiceComponent> permittedServiceComponents = serviceRolesToServiceComponents(serviceRoleGroup.getServiceRoles());
+        Set<ServiceComponent> permittedComponents = serviceRolesToServiceComponents(serviceRoleGroup.getRoles());
+        for (ServiceComponent serviceComponent: serviceComponents) {
+            if (!permittedServiceComponents.contains(serviceComponent) &&
+                !permittedComponents.contains(ServiceComponent.of(null, serviceComponent.getComponent()))) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private boolean areRequiredServiceRolesPresent(ServiceRoleGroup serviceRoleGroup, Set<ServiceComponent> serviceComponents) {
+        boolean requiredServiceRolesPresent = serviceComponents
+                .containsAll(serviceRolesToServiceComponents(serviceRoleGroup.getRequiredServiceRoles()));
+        boolean requiredRolesPresent = serviceComponents.stream()
+                .map(x -> ServiceComponent.of(null, x.getComponent()))
+                .collect(Collectors.toSet())
+                .containsAll(serviceRolesToServiceComponents(serviceRoleGroup.getRequiredRoles()));
+        return requiredServiceRolesPresent && requiredRolesPresent;
+    }
+
+    private Set<ServiceComponent> serviceRolesToServiceComponents (Set<ServiceRoleGroup.ServiceRole> serviceRoles) {
+        return serviceRoles.stream()
+                .map(x -> ServiceComponent.of(x.getService(), x.getRole()))
+                .collect(Collectors.toSet());
+    }
+
     private boolean hasInstanceGroupAwsEphemeralStorage(InstanceGroup instanceGroup) {
         return instanceGroup.getTemplate().getVolumeTemplates().stream().anyMatch(volume -> AwsDiskType.Ephemeral.value().equals(volume.getVolumeType()));
     }
 
-    private boolean onlyEphemeralStorageInInstanceGroup(InstanceGroup instanceGroup) {
-        return instanceGroup.getTemplate().getVolumeTemplates().stream().allMatch(volume -> AwsDiskType.Ephemeral.value().equals(volume.getVolumeType()));
+    @VisibleForTesting
+    boolean isInstanceGroupEphemeralVolumesOnly (InstanceGroup ig) {
```

Comment:
this method is a duplicate from `StackCreationService`, so the code smell seems to be valid and this should be refactored